components-api/src/main/scala/pl/touk/nussknacker/engine/api/context/transformation/GenericNodeTransformation.scala

```diff
@@ -20,7 +21,7 @@ import pl.touk.nussknacker.engine.api.typed.typing.{TypingResult, Unknown}
      - for sources OutputVariable *has* to be used for Flink sources, it's value is always equal to 'input' ATM, due to source API limitations
  */
 // TODO: rename to DynamicComponent
-sealed trait GenericNodeTransformation[T] {
+sealed trait GenericNodeTransformation[T] extends Component {
```

Comment:
Because, now after extracting a separate domain model class for global variables, we can have a `Component` instead of `Any` class in our domain object class in the field `ComponentDefinitionWithImplementation.implementation` and `GenericNodeTransformation` in the `DynamicComponentDefinitionWithImplementation.implementation`. This class was always designed to be mixed into `Component` (`SourceFactory`, `Service`, etc.). I wanted to add a restriction instead of inheritance (`{ self => Component`) but for some reason, pattern matching on this class is not usable.